dex/pair/src/safe_price.rs

```diff
@@ -100,7 +105,14 @@ pub trait SafePriceModule:
             new_index = (safe_price_current_index % MAX_OBSERVATIONS) + 1;
         }
 
-        if last_price_observation.recording_round == current_round {
+        let rounds_since_last_observation = current_round - last_price_observation.recording_round;
```

Comment:
I'm not sure if this is correct for the very first observation, as that will always have round 0.